sw/airborne/firmwares/fixedwing/guidance/guidance_v.c

```diff
@@ -251,6 +308,50 @@ void v_ctl_climb_loop(void)
   }
 }
 
+void v_ctl_landing_loop(void)
+{
+  static float land_speed_i_err;
+  static float land_alt_i_err;
+  static float kill_alt;
+  float land_speed_err = v_ctl_landing_desired_speed - stateGetHorizontalSpeedNorm_f();
+  float land_alt_err = v_ctl_altitude_setpoint - stateGetPositionUtm_f()->alt;
+
+  if (kill_throttle
+      && (kill_alt - v_ctl_altitude_setpoint)
+          > (v_ctl_landing_alt_throttle_kill - v_ctl_landing_alt_flare)) {
+    v_ctl_throttle_setpoint = 0.0;  // Throttle is already in KILL (command redundancy)
+    nav_pitch = v_ctl_landing_pitch_flare;  // desired final flare pitch
+    lateral_mode = LATERAL_MODE_ROLL;  //override course correction during flare - eliminate possibility of catching wing tip due to course correction
+    h_ctl_roll_setpoint = 0.0;  // command zero roll during flare
+  } else {
+    // set throttle based on altitude error
+    v_ctl_throttle_setpoint = land_alt_err * v_ctl_landing_throttle_pgain
+        + land_alt_i_err * v_ctl_landing_throttle_igain;
+    Bound(v_ctl_throttle_setpoint, 0, v_ctl_landing_throttle_max * MAX_PPRZ);  // cut off throttle cmd at specified MAX
+
+    land_alt_i_err += land_alt_err / CONTROL_FREQUENCY;  // integrator land_alt_err, divide by control frequency
+    BoundAbs(land_alt_i_err, 50);  // integrator sat limits
+
+    // set pitch based on airspeed error
```

Comment:
here, ground speed is used, not airspeed, but it is maybe only an error in the comment